src/vs/workbench/contrib/chat/browser/promptSyntax/pickers/promptFilePickers.ts

```diff
@@ -70,12 +73,65 @@ export interface ISelectPromptResult {
 /**
  * Button that opens the documentation.
  */
-const HELP_BUTTON: IQuickInputButton = Object.freeze({
-	tooltip: localize('help', "Help"),
-	iconClass: ThemeIcon.asClassName(Codicon.question),
-});
+function newHelpButton(type: PromptsType): IQuickInputButton & { helpURI: URI } {
+	const iconClass = ThemeIcon.asClassName(Codicon.question);
+	switch (type) {
+		case PromptsType.prompt:
+			return {
+				tooltip: localize('help.prompt', "Show help on prompt files"),
+				helpURI: URI.parse(PROMPT_DOCUMENTATION_URL),
+				iconClass
+			};
+		case PromptsType.instructions:
+			return {
+				tooltip: localize('help.instructions', "Show help on instruction files"),
+				helpURI: URI.parse(INSTRUCTIONS_DOCUMENTATION_URL),
+				iconClass
+			};
+		case PromptsType.agent:
+			return {
+				tooltip: localize('help.agent', "Show help on custom agent files"),
+				helpURI: URI.parse(AGENT_DOCUMENTATION_URL),
+				iconClass
+			};
+	}
+}
+
+function isHelpButton(button: IQuickInputButton): button is IQuickInputButton & { helpURI: URI } {
+	return 'helpURI' in button;
+}
```

Comment:
Don't we ban this via eslint now due to the possibility of it breaking mangling? I thought the preferred was `(<InterfaceWIthIt>button).helpURI`?